receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -579,10 +593,29 @@ func (c *tracesConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSe
 			err = c.nextConsumer.ConsumeTraces(session.Context(), traces)
 			c.obsrecv.EndTracesOp(ctx, c.unmarshaler.Encoding(), spanCount, err)
 			if err != nil {
-				if c.messageMarking.After && c.messageMarking.OnError {
-					session.MarkMessage(message, "")
+				if errorRequiresBackoff(err) && c.backOff != nil {
+					backOffDelay := c.backOff.NextBackOff()
+					if backOffDelay == backoff.Stop {
+						if c.messageMarking.After && c.messageMarking.OnError {
+							session.MarkMessage(message, "")
+						}
+						return err
+					}
```

Comment:
If max elapsed time of backoff config is exceeded, mark the message and return the error.